examples/addsvc/cmd/addcli/addcli.go

```diff
@@ -54,35 +57,55 @@ func main() {
 
 	// This is a demonstration client, which supports multiple tracers.
 	// Your clients will probably just use one tracer.
-	var tracer stdopentracing.Tracer
+	var otTracer stdopentracing.Tracer
 	{
-		if *zipkinURL != "" {
-			collector, err := zipkin.NewHTTPCollector(*zipkinURL)
+		if *zipkinV1URL != "" {
+			collector, err := zipkinot.NewHTTPCollector(*zipkinV1URL)
 			if err != nil {
 				fmt.Fprintln(os.Stderr, err.Error())
 				os.Exit(1)
 			}
 			defer collector.Close()
 			var (
 				debug       = false
-				hostPort    = "localhost:80"
-				serviceName = "addsvc"
+				hostPort    = "localhost:0"
+				serviceName = "addsvc-cli"
 			)
-			recorder := zipkin.NewRecorder(collector, debug, hostPort, serviceName)
-			tracer, err = zipkin.NewTracer(recorder)
+			recorder := zipkinot.NewRecorder(collector, debug, hostPort, serviceName)
+			otTracer, err = zipkinot.NewTracer(recorder)
 			if err != nil {
 				fmt.Fprintln(os.Stderr, err.Error())
 				os.Exit(1)
 			}
 		} else if *lightstepToken != "" {
-			tracer = lightstep.NewTracer(lightstep.Options{
+			otTracer = lightstep.NewTracer(lightstep.Options{
 				AccessToken: *lightstepToken,
 			})
-			defer lightstep.FlushLightStepTracer(tracer)
+			defer lightstep.FlushLightStepTracer(otTracer)
 		} else if *appdashAddr != "" {
-			tracer = appdashot.NewTracer(appdash.NewRemoteCollector(*appdashAddr))
+			otTracer = appdashot.NewTracer(appdash.NewRemoteCollector(*appdashAddr))
 		} else {
-			tracer = stdopentracing.GlobalTracer() // no-op
+			otTracer = stdopentracing.GlobalTracer() // no-op
+		}
+	}
+
+	var zipkinTracer *zipkin.Tracer
+	{
+		var (
+			err         error
+			hostPort    = "" // if host:port is unknown we can keep this empty
+			serviceName = "addsvc-cli"
+		)
+		noopTracer := (*zipkinV2URL == "")
```

Comment:
Just for readability purposes I'd call it `useNoopTracer`. It took me a couple of seconds to realize what was the intention in line 104.

Reply:
agreed... will update